src/main/scala/com/cloudera/sparkts/DateTimeIndex.scala

```diff
@@ -240,48 +255,136 @@ object DateTimeIndex {
   }
 
   /**
-   * Create a UniformDateTimeIndex with the given start time, number of periods, and frequency.
+   * Create a UniformDateTimeIndex with the given start time, number of periods, frequency
+   * and time zone.
+   */
+  def uniform(start: Long, periods: Int, frequency: Frequency, zone: DateTimeZone)
+  : UniformDateTimeIndex = {
+    new UniformDateTimeIndex(start, periods, frequency, zone)
+  }
+
+  /**
+   * Create a UniformDateTimeIndex with the given start time, number of periods, and frequency
+   * using the time zone of start time.
    */
   def uniform(start: DateTime, periods: Int, frequency: Frequency): UniformDateTimeIndex = {
-    new UniformDateTimeIndex(start.getMillis, periods, frequency)
+    new UniformDateTimeIndex(start.getMillis, periods, frequency, start.getZone)
+  }
+
+  /**
+   * Create a UniformDateTimeIndex with the given start time, number of periods, frequency
+   * , and time zone
+   */
+  def uniform(start: DateTime, periods: Int, frequency: Frequency, zone: DateTimeZone)
+  : UniformDateTimeIndex = {
+    new UniformDateTimeIndex(start.getMillis, periods, frequency, zone)
   }
 
   /**
    * Create a UniformDateTimeIndex with the given start time and end time (inclusive) and frequency.
    */
   def uniform(start: Long, end: Long, frequency: Frequency): UniformDateTimeIndex = {
-    uniform(
-      start, frequency.difference(new DateTime(start, UTC), new DateTime(end, UTC)) + 1, frequency)
+    uniform(start, frequency.difference(new DateTime(start, DateTimeZone.getDefault()),
```

Comment:
Maybe put `val tz = DateTimeZone.getDefault()` and use `tz` in the three places here?